gadgets/snapshot_file/README.mdx

```diff
@@ -0,0 +1,128 @@
+---
+title: snapshot_file
+sidebar_position: 0
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+The `snapshot_file` gadget shows open files for running processes.
+
+## Getting started
+
+Running the gadget:
+
+<Tabs groupId="env">
+    <TabItem value="kubectl-gadget" label="kubectl gadget">
+        ```bash
+        $ kubectl gadget run ghcr.io/inspektor-gadget/gadget/snapshot_file:%IG_TAG% [flags]
+        ```
+    </TabItem>
+
+    <TabItem value="ig" label="ig">
+        ```bash
+        $ sudo ig run ghcr.io/inspektor-gadget/gadget/snapshot_file:%IG_TAG% [flags]
+        ```
+    </TabItem>
+</Tabs>
+
+## Flags
+
+### `--paths`
+
+Show full file paths (by default, only dentry names are shown).
+
+Default value: `"false"`
+
+### `--file-type-mask`
+
+Filter which file types to display using a bitmask.  
+By default, only **regular files** and **directories** are shown.
+
+Supported file types:
+
+- Regular file
+- Directory
+- Pipe
+- Socket
+- BPF map
+- Character device
+- Block device
+- Symlink
+- Other
+
+## Guide
+
+Run a pod / container:
+
+<Tabs groupId="env">
+<TabItem value="kubectl-gadget" label="kubectl gadget">
+```bash
+$ kubectl run --restart=Never --image=busybox test-snapshot-file -- sh -c 'sleep inf'
+pod/test-snapshot-file created
+````
+
+</TabItem>
+
+<TabItem value="ig" label="ig">
+```bash
+$ docker run --name test-snapshot-file -d busybox /bin/sh -c 'sleep inf'
+...
+```
+</TabItem>
+</Tabs>
+
+Then, open a file in the container:
+
+<Tabs groupId="env">
+<TabItem value="kubectl-gadget" label="kubectl gadget">
+```bash
+$ kubectl exec -it test-snapshot-file -- sh -c 'echo hello > /tmp/demo.txt && sleep inf'
```

Comment:
The file handle should be directly closed after the echo exits? (Maybe not?) So its hard to run the snapshot gadget in that exact millisecond.

You could use `tail -f <some textfile` to keep an open FD. Sleep won't be needed then too